pkg/kv/kvserver/rangefeed/budget.go

```diff
@@ -118,16 +118,20 @@ type FeedBudget struct {
 	limit int64
 	// Channel to notify that memory was returned to the budget.
 	replenishC chan interface{}
-	// Budget cancellation request
+	// Budget cancellation request.
 	stopC chan interface{}
+	// Atomic value reflecting RangefeedBudgetsEnabled cluster setting. When
+	// budgets are disabled, all new messages should pass through, all previous
+	// allocations should still be returned as memory pool is shared with SQL.
+	budgetsEnabled *int32
```

Comment:
I wonder though: is this any faster than simply checking the setting directly? That essentially does the same thing: checks an atomic int, with an additional map lookup.

https://github.com/cockroachdb/cockroach/blob/5b5fcee012ad2307e1b01748f44e865fd1a951f3/pkg/settings/values.go#L82

Reply:
Lookup in settings:
cpu: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
BenchmarkTryGet
BenchmarkTryGet-16      424617386            2.820 ns/op

BenchmarkTryGet
BenchmarkTryGet-16      441149536            2.649 ns/op

BenchmarkTryGet
BenchmarkTryGet-16      445762287            2.618 ns/op

Lookup via dedicated atomic:
cpu: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
BenchmarkTryGet
BenchmarkTryGet-16      440164717            2.642 ns/op

BenchmarkTryGet
BenchmarkTryGet-16      458073957            2.612 ns/op

BenchmarkTryGet
BenchmarkTryGet-16      455464820            2.598 ns/op

Looks like extra atomic is not worth the effort, we can just lookup directly and map lookup is fast.
Those are microbenchmarks, dunno if it'll have any effects on caches or anything, but we must be accessing settings pretty regularly so likely not too bad.

Reply:
Interestingly, adding a nil check for settings prior to reading them (to be able to not provide settings in tests) adds more to the bench time than switching from atomic to settings reads.

Reply:
Nice, that's what I thought -- direct settings lookup is fine here.

I've also been surprised by how expensive those `nil` checks can be, I've found an extra `bool` to signal presence to be much more performant. Anyway, requiring settings in tests seems fine.